src/charm.py

```diff
@@ -131,22 +137,18 @@ def _on_update_status(self, _: EventBase) -> None:  # noqa: C901
             self.model.unit.status = BlockedStatus("Cannot relate to more than one grafana-agent")
             return
 
-        config_valied, confg_valid_message = self.validate_exporter_configs()
-        if not config_valied:
-            self.model.unit.status = BlockedStatus(confg_valid_message)
-            return
-
         hw_tool_ok, error_msg = self.hw_tool_helper.check_installed()
         if not hw_tool_ok:
             self.model.unit.status = BlockedStatus(error_msg)
             return
 
-        if not self.exporter.check_health():
-            logger.warning("Exporter health check - failed.")
-            restart_ok, restart_status = self.restart_exporter()
-            if not restart_ok and restart_status is not None:
-                self.model.unit.status = restart_status
-                return
+        if self._stored.exporter_installed:  # type: ignore[truthy-function]
+            if not self.exporter.check_active():
```

Comment:
```suggestion
            if not self.exporter.check_health():
```
Wrong function being called. Will make this change.